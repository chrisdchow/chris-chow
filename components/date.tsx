import { parseISO, format } from 'date-fns';
import { FunctionComponent } from 'react';

type DateProps = {
  dateString: string;
};

export const Date: FunctionComponent<DateProps> = ({ dateString }) => {
  const date = parseISO(dateString);
  return <time dateTime={dateString}>{format(date, 'LLLL d, yyyy')}</time>;
};
