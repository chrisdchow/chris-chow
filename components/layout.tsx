import { FunctionComponent } from 'react';
import Head from 'next/head';
import Link from 'next/link';
import { H1, H2 } from '@components/elements/heading';

const name = 'Chris Chow';
export const siteTitle = name;

type LayoutProps = {
  home?: boolean;
};

export const Layout: FunctionComponent<LayoutProps> = ({ home, children }): JSX.Element => {
  const homeHeader = (
    <>
      <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
      <H1>{name}</H1>
    </>
  );

  const navigateHomeHeader = (
    <>
      <Link href='/'>
        <a>
          <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
        </a>
      </Link>
      <H2>
        <Link href='/'>
          <a>{name}</a>
        </Link>
      </H2>
    </>
  );

  return (
    <div className='max-w-xl py-0 px-4 mx-auto my-12'>
      <Head>
        <link rel='icon' href='/favicon.ico' />
        <meta name='description' content='Chris Chow' />
        <meta property='og:image' content={`https://og-image.now.sh/${encodeURI(siteTitle)}.png`} />
        <meta name='og:title' content={siteTitle} />
        <meta name='twitter:card' content='summary_large_image' />
      </Head>
      <header className='flex flex-col items-center'>{home ? homeHeader : navigateHomeHeader}</header>
      <main>{children}</main>
      {!home && (
        <div className='mt-12 mx-0 mb-0'>
          <Link href='/'>
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  );
};
