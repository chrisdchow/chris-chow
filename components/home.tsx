import { Layout, siteTitle } from '@components/layout';
import { FunctionComponent } from 'react';
import Head from 'next/head';
import Link from 'next/link';
import { Date } from '@components/date';
import utilStyles from '@styles/utils.module.css';

type HomeProps = {
  posts: {
    id: string;
    date: any;
    title: string;
  }[];
};

export const Home: FunctionComponent<HomeProps> = ({ posts }): JSX.Element => (
  <Layout home>
    <Head>
      <title>{siteTitle}</title>
    </Head>
    <div className='space-y-6'>
      <section>
        <p>
          Philadelphia born and raised software engineer currently based in San
          Francisco. You may view my resume{' '}
          <a href='/christopher-chow-resume.pdf' target='_blank'>
            here
          </a>
          .
        </p>
      </section>
      <section>
        <h2 className=''>Posts</h2>
        <ul className='list-none space-y-4'>
          {posts.map(({ id, date, title }) => (
            <li className='' key={id}>
              <Link href='/posts/[id]' as={`/posts/${id}`}>
                <a>{title}</a>
              </Link>
              <br />
              <small className={utilStyles.lightText}>
                <Date dateString={date} />
              </small>
            </li>
          ))}
        </ul>
      </section>
    </div>
  </Layout>
);
