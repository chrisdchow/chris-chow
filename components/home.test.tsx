import React from 'react';
import { render } from '@test/test-utils';
import { Home } from './home';

describe('Home page', () => {
  it('matches snapshot', () => {
    const posts = [];
    const projects = [];

    const { asFragment } = render(<Home posts={posts} projects={projects} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
