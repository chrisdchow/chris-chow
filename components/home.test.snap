// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <img
        alt="Chris Chow"
        class="headerHomeImage borderCircle"
        src="/images/profile.jpg"
      />
      <h1
        class="heading2Xl"
      >
        Chris Chow
      </h1>
    </header>
    <main>
      <section
        class="headingMd"
      >
        <p>
          [Your Self Introduction]
        </p>
        <p>
          (This is a sample website - youâ€™ll be building a site like this on 
          <a
            href="https://nextjs.org/learn"
          >
            our Next.js tutorial
          </a>
          .)
        </p>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
